inc/wp-custom-menu.php

```diff
@@ -8,6 +8,355 @@
+					$dropdown[ $m->ID ]['href']        = $m->url;
+					$dropdown[ $m->ID ]['has_current'] = false;
+					$dropdown[ $m->ID ]['parent']      = $m->menu_item_parent;
+					$dropdown[ $m->ID ]['items']       = array();
+					$dropdown[ $m->ID ]['cta_button']  = get_field( 'menu_cta_button', $m );
+					$dropdown[ $m->ID ]['cta_color'] = get_field( 'menu_cta_button_color', $m );
+
+
+
+					// The menu link can be relative or absolute.
+					// Format menu link and remove absolute base url from link
+					$prefix = get_home_url();
+					$menu_url = rtrim( $m->url, '/' ) . '/';
+					if ( 0 === strpos( $menu_url, $prefix ) ) {
+							$menu_url = substr( $menu_url, strlen( $prefix ) );
+					}
+					$menu_url = $subsite_base_folder . $menu_url;
+
+					if ( $current_uri === $menu_url ) {
+						$pre_menu[ $m->menu_item_parent ]['has_current'] = true;
+					}
+
+					/**
+					 * Add the current child's data to the existing $pre_menu array under 'items',
+					 * and then under this item's ID, for that parent ID
+					 */
+					if($dropdown[ $m->ID ]['type'] == 'button'){
+						
+						$pre_menu[ $m->menu_item_parent ]['cta-buttons'][ $m->ID ] = $dropdown[ $m->ID ];
+												
+					} else {
+						$pre_menu[ $m->menu_item_parent ]['items'][ $m->ID ] = $dropdown[ $m->ID ];
+						
+					}
+						
+
+					
+				}
+			}
+
+			/**
+			 * Step 3: Loop through every source menu item a third time. If this item has a
+			 * parent value, but that value IS NOT IN the top-level menu array, build an array
+			 * of data for this menu item
+			 */
+			$column = array();
+			$our_array_menu = array();
+			foreach ( $array_menu as $m ) {
+				if ( $m->menu_item_parent && ! array_key_exists( $m->menu_item_parent, $pre_menu ) ) {
+
+					$column[ $m->ID ]                = array();
+					$column[ $m->ID ]['ID']          = $m->ID;
+					$column[ $m->ID ]['type']        = get_field( 'uds_menu_item_type', $m );
+					$column[ $m->ID ]['cta_color'] = get_field( 'menu_cta_button_color', $m );
+					$column[ $m->ID ]['text']        = $m->title;
+					$column[ $m->ID ]['href']        = $m->url;
+					$column[ $m->ID ]['has_current'] = false;
+
+					/**
+					 * Add this item's data as a child to the $dropdown array we created in step 2.
+					 * Place it under the parent, then under 'items', in a new array with ID of this item's ID.
+					 */
+					$dropdown[ $m->menu_item_parent ]['items'][ $m->ID ] = $column[ $m->ID ];
+					
+					/**
+					 * Determine this item's top-menu item (grandparent) by getting the parent ID of this item's parent.
+					 * Adding a check here to ensure that there is a parent array in the parent of this item for us to
+					 * add anything to.
+					 */
+					if ( array_key_exists( 'parent', $dropdown[ $m->menu_item_parent ] ) ) {
+						$top_menu = $dropdown[ $m->menu_item_parent ]['parent'];
+			 /* At step 3, if menu items are still leftover, we have a column header above our menu items
```

Comment:
These lines here looked odd and at various points produced a lot of PHP warnings. Upon a closer inspection, I think that this may have been inadvertently pasted or included from another version of this walker script. 
- The variable `$our_array_menu` is never referenced elsewhere in the code.
- `$middle_menu` is an array here not an object.

And the most telling: If you comment this whole section out, things still appear to function normally. 😄 

The logic in line 248 is just about the same as the logic in line 231. Pretty sure that the second section here beginning at line 235 is the correct logic for getting those grandchildren into the correct place.